api/grpc_server.go

```diff
@@ -226,8 +228,37 @@ type TxAPI interface {
 	ValidatedLayer() types.LayerID
 }
 
+// create new NodeConfigParams object from Config params
+func newNodeConfigParams(config *cfg.Config) *pb.NodeConfigParams {
+	return &pb.NodeConfigParams{
+		TestMode:               config.TestMode,
+		LayerDurationSec:       uint32(config.LayerDurationSec),
+		LayerAvgSize:           uint32(config.LayerAvgSize),
+		LayersPerEpoch:         uint32(config.LayersPerEpoch),
+		Hdist:                  uint32(config.Hdist),
+		PoETServer:             config.PoETServer,
+		SyncRequestTimeout:     uint32(config.SyncRequestTimeout),
+		AtxsPerBlock:           uint32(config.AtxsPerBlock),
+		Bootstrap:              config.P2P.SwarmConfig.Bootstrap,
+		BootstrapNodes:         config.P2P.SwarmConfig.BootstrapNodes,
+		Randcon:                uint32(config.P2P.SwarmConfig.RandomConnections),
+		MaxInbound:             uint32(config.P2P.MaxInboundPeers),
+		HareWakeupDelta:        uint32(config.HARE.WakeupDelta),
+		HareRoundDuration:      uint32(config.HARE.RoundDuration),
+		HareCommitteeSize:      uint32(config.HARE.N),
+		HareMaxAdversaries:     uint32(config.HARE.F),
+		EligibilityConf:        uint32(config.HareEligibility.ConfidenceParam),
+		EligibilityEpochOffset: uint32(config.HareEligibility.EpochOffset),
+		GenesisActiveSize:      uint32(config.GenesisActiveSet),
+		PostSize:               config.POST.SpacePerUnit,
+		PostLabels:             config.POST.LabelsLogRate,
+		GenesisTime:            config.GenesisTime,
+		NetworkId:              uint32(config.P2P.NetworkID),
+	}
```

Comment:
looks good in terms of the params
I would run a test with this branch to see if it achieves what you need before merging though